File: SDKs/amazonaws.com/kinesis/2013-12-02/go/pkg/models/operations/enableenhancedmonitoring.go
package operations

import (
	"openapi/pkg/models/shared"
)

type EnableEnhancedMonitoringXAmzTargetEnum string

const (
	EnableEnhancedMonitoringXAmzTargetEnumKinesis20131202EnableEnhancedMonitoring EnableEnhancedMonitoringXAmzTargetEnum = "Kinesis_20131202.EnableEnhancedMonitoring"
)

type EnableEnhancedMonitoringHeaders struct {
	XAmzAlgorithm     *string                                `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                                `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                                `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                                `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                                `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                                `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                                `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        EnableEnhancedMonitoringXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type EnableEnhancedMonitoringRequest struct {
	Headers EnableEnhancedMonitoringHeaders
	Request shared.EnableEnhancedMonitoringInput `request:"mediaType=application/json"`
}

type EnableEnhancedMonitoringResponse struct {
	ContentType               string
	EnhancedMonitoringOutput  *shared.EnhancedMonitoringOutput
	InvalidArgumentException  *interface{}
	LimitExceededException    *interface{}
	ResourceInUseException    *interface{}
	ResourceNotFoundException *interface{}
	StatusCode                int64
}

File: SDKs/amazonaws.com/kinesis/2013-12-02/typescript/src/sdk/models/operations/enableenhancedmonitoring.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum EnableEnhancedMonitoringXAmzTargetEnum {
    Kinesis20131202EnableEnhancedMonitoring = "Kinesis_20131202.EnableEnhancedMonitoring"
}


export class EnableEnhancedMonitoringHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: EnableEnhancedMonitoringXAmzTargetEnum;
}


export class EnableEnhancedMonitoringRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: EnableEnhancedMonitoringHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.EnableEnhancedMonitoringInput;
}


export class EnableEnhancedMonitoringResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  enhancedMonitoringOutput?: shared.EnhancedMonitoringOutput;

  @SpeakeasyMetadata()
  invalidArgumentException?: any;

  @SpeakeasyMetadata()
  limitExceededException?: any;

  @SpeakeasyMetadata()
  resourceInUseException?: any;

  @SpeakeasyMetadata()
  resourceNotFoundException?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}
